api/testdata/api.snapshot

```diff
@@ -5062,12 +5062,12 @@ HTTP/1.1 200 OK
 Connection: close
 
 /* snapshot: TestFSIHandlers init case 0: POST / */
-HTTP/1.1 500 Internal Server Error
+HTTP/1.1 400 Bad Request

```

Comment:
This case means we don't properly parse/unmarshal the structs or the test is set up wrong. Would be nice to actually fix it as these will be useful for keeping record/testing of how things work until we transition to the new JSON based test framework.

Reply:
In `InitDataset`, the first place we check the Ref is `dsref.IsValidName`, and return `dsref.ErrDescribeValidName`. This test is supposed to be checking that we error when there is no Ref. The status code change occurred because `NewHTTPRequestMethod` now uses `RespondWithErr` rather than `WriteErrResponse`, with every error returning `500`.

We should probably have a `dsref.ErrEmtpyRef` error and `InitDataset` should be responding with that instead (and a `422` error probably?).

In `RespondWithErr` we have `dsref.ErrBadCaseShouldRename`, `dsref.ErrDescribeValidName`, and `dsref.ErrDescribeValidUsername`. Do those make sense as `400 Bad Request` errors, or should they be `422 Unprocessable Entity`?